README.md

```diff
@@ -0,0 +1,68 @@
+# Accompanying code for Unit Testing C++ course
+
+## Prerequisites
+
+In order to run the examples in this repository, you need a C++ compiler (supporting
+C++14 or newer) and CMake (version 3.11 or newer).
+
+On Ubuntu, you can install your dependencies like this:
+
+```sh
+sudo apt install build-essential cmake
+```
+
+On macOS, you can install them with [Homebrew](https://brew.sh/):
+
+```sh
+brew install cmake
+```
+
+If you are using Visual Studio on Windows, you do not require any additional
+dependencies.
+
+## Building projects from the command line
+
+If you are doing this with Visual Studio on Windows, you will need to open the Visual
+Studio Developer Shell for this step.
+
+First, navigate to the project you wish to build, e.g.:
+
+```sh
+cd src/Chapter2
+```
+
+(If you wish to build all projects, you can just invoke CMake from the root directory of
+this repository.)
+
+Then you can build using CMake in the usual way:
+
+```sh
+cmake -B build
+cmake --build build/
+```
+
+## Running tests from the command line
+
+Once you have built your project, you can then run the tests by using [CTest], which is
```

Comment:
It is unclear to me what's the difference between `gtest` and `CTest`. Given that we talk about `gtest` in the course itself, it might confusing for students to come here and find out they need to use `CTest` instead unless some explanation is given.

Reply:
I probably have confused matters a bit here. I'm still using GoogleTest (`gtest`) in this repo though!

CTest is a part of CMake for integrating unit tests, so you don't have to execute the test program directly and your IDE can easily find it etc. I think it might also be a test framework in its own right, but that's not how we're using it here.

Reply:
Maybe mentioning you can call the compiled google tests directly, without using (or if you don't have access to) CTest.

Reply:
@jamesturner246 Good point. In a way, using CTest rather than invoking the test binary directly means you end up glossing over that detail, which is probably an important one. I was thinking that maybe we could talk students through this in the course though, rather than in the instructions.

E.g., something like:

> After you compile your program, you'll notice that there's another binary called `Chapter2_test` in folder [whatever]. If you run this, you can see whether your tests are passing or failing.
>
> As we are using CMake though, you can also just run `ctest`, which will run your test binary (or binaries) for the project -- whatever they are called -- and display a summary of the results. This is also the way that many IDEs will run your tests, if they have this feature.

What does everyone think?

Reply:
Fair enough, that also works